README.md

```diff
@@ -1,70 +1,363 @@
+subproject if there is an existing `build/` subdirectory, and if there is, just
+call `make`. If there is no `build/` subdirectory, it will call `make -f
+Makefile.repo build` instead. Notice that this handles each subproject
+individually, so you can have a mixture of existing and non-existing `build/`
+subdirectories.
+
+    ./build-all
+
+This builds everything from scratch.
+
+Now continue editing files in subprojects and then
+
+    ./build-all
+
+This now just invokes `make` in each subproject, keeping the object
+files for unchanged sources.
+
+### make install
+
+Of course you need root permissions to install any file to the system:
+
+    ./build-all
+    sudo ./build-all install
+
+
+### Keeping Changes Locally in the Source Tree
+
+The CMake environments in the subprojects are set up to prefer header files and
+built libraries from sibling subprojects over those from the system; so you can
+work for a long time in the source tree without a need for `sudo make install`.
+
+For the slightly paranoid:
+
+To make absolutely sure that you are only using headers and libs from your
+source tree, you can move the headers away in the system directories while you
+develop and later move them back:
+
+    cd /usr/include
+    sudo mv yui yui.orig
+
+    cd /usr/lib64
+    sudo mkdir libyui.save
+    sudo mv libyui.so* libyui.save
+    sudo mv yui libyui.save
+
+Make sure to restore this later when you are done:
+
+    cd /usr/include
+    sudo rm -rf yui
+    sudo mv yui.orig yui
+
+    cd /usr/lib64
+    sudo rm -rf libyui.so* yui
+    sudo mv libyui.save/* .
+    sudo rmdir libyui.save
+
+_If you wrecked it, force-reinstall the libyui packages:_
+
+    rpm -qa "libyui*" >/tmp/libyui-pkg
+    sudo zypper install --force $(cat /tmp/libyui-pkg)
+
+
+# Version Numbers
+
+All packages in this source tree have the same version numer; that's why the
+subprojects share the toplevel `VERSION.cmake` file.
+
+The general idea is to enable transaction-like changes on the libyui base
+packages to avoid long delays in the distribution build cycle and broken builds:
+
+Some parts may have become incompatible, yet the higher-level parts require the
+base lib parts to be published as a prerequisite for building. In the past,
+this resulted in staging projects not building; manual interaction was often
+needed to break depencency cycles.
+
+The version numbers are also in all the .spec files in the `package/`
+subdirectory, so they need to be kept consistent among each other and with the
+`VERSION.cmake` file.
+
+For a simple version number increment, use
+
+    rake version:bump
+
+that does the required changes consistently.
+
+
+This requires some more packages to be installed:
+
+- rake (part of the ruby base package on SUSE distributions)
+- the packaging_rake_tasks ruby gem
+- the libyui-rake ruby gem
+
+At the time of this writing, those packages were named
+
+- ruby2.7
+- ruby2.7-rubygem-libyui-rake
+- ruby2.7-rubygem-packaging_rake_tasks
```

Comment:
No way. I lost count how many times I was cursing those packages because of their crazy (and constantly changing) naming scheme. I always have to hunt that stuff down, and it's always a PITA.